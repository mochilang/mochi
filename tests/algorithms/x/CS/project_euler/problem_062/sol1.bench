{"name":"main","duration_us":17388,"memory_bytes":20973144}