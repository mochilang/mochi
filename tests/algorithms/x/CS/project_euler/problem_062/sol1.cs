// Generated by Mochi 0.10.60 on 2025-08-08 11:10 +0700
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json;
using System.Numerics;
using System.Collections;

class Program {
    static bool seededNow = false;
    static long nowSeed = 0;
    static long _now() {
        if (!seededNow) {
            var s = Environment.GetEnvironmentVariable("MOCHI_NOW_SEED");
            if (long.TryParse(s, out var v)) {
                nowSeed = v;
                seededNow = true;
            }
        }
        if (seededNow) {
            nowSeed = unchecked(nowSeed * 1664525 + 1013904223);
            nowSeed %= 9223372036854775783L;
            if (nowSeed < 0) nowSeed += 9223372036854775783L;
            return nowSeed;
        }
        return DateTime.UtcNow.Ticks / 100;
    }
    static long _mem() {
        return GC.GetTotalAllocatedBytes(true);
    }
    static long _len(object v) {
        if (v is Array a) return a.Length;
        if (v is string s) return s.Length;
        if (v is System.Collections.ICollection c) return c.Count;
        return Convert.ToString(v).Length;
    }
    static long _atoi(object v) {
        if (v == null) return 0;
        if (v is long l) return l;
        if (v is int i) return i;
        if (v is double d) return (long)d;
        if (v is bool b) return b ? 1L : 0L;
        if (v is string s) {
            if (long.TryParse(s, out var n)) return n;
            if (double.TryParse(s, out var f)) return (long)f;
            return 0;
        }
        try { return Convert.ToInt64(v); } catch { return 0; }
    }
    static string _substr(string s, long start, long end) {
        if (start < 0) start = 0;
        if (end < 0) end = 0;
        if (start > s.Length) start = s.Length;
        if (end > s.Length) end = s.Length;
        if (start > end) start = end;
        return s.Substring((int)start, (int)(end - start));
    }
    static string _fmt(object v) {
        if (v is Array a) {
            var parts = new List<string>();
            foreach (var x in a) parts.Add(_fmt(x));
            return "[" + string.Join(", ", parts) + "]";
        }
        if (v is System.Collections.IDictionary d) {
            var keys = new List<string>();
            foreach (var k in d.Keys) keys.Add(k.ToString());
            keys.Sort();
            var parts = new List<string>();
            foreach (var k in keys) parts.Add(k + ":" + _fmt(d[k]));
            return "map[" + string.Join(", ", parts) + "]";
        }
        if (v is System.Collections.IEnumerable e && !(v is string)) {
            var parts = new List<string>();
            foreach (var x in e) parts.Add(_fmt(x));
            return string.Join(", ", parts);
        }
        if (v is string s) return "\"" + s.Replace("\"", "\\\"") + "\"";
        if (v is bool b) return b ? "true" : "false";
        return Convert.ToString(v);
    }
    static string _fmtStr(object v) {
        if (v is Array a) {
            var parts = new List<string>();
            foreach (var x in a) parts.Add(_fmtStr(x));
            return "[" + string.Join(" ", parts) + "]";
        }
        if (v is System.Collections.IDictionary d) {
            var keys = new List<string>();
            foreach (var k in d.Keys) keys.Add(k.ToString());
            keys.Sort();
            var parts = new List<string>();
            foreach (var k in keys) parts.Add(k + ":" + _fmtStr(d[k]));
            return "map[" + string.Join(" ", parts) + "]";
        }
        if (v is System.Collections.IEnumerable e && !(v is string)) {
            var parts = new List<string>();
            foreach (var x in e) parts.Add(_fmtStr(x));
            return string.Join(" ", parts);
        }
        if (v is string s) return "\"" + s.Replace("\"", "\\\"") + "\"";
        if (v is bool b) return b ? "true" : "false";
        return Convert.ToString(v);
    }
    public static string _fmtTop(object v) {
        if (v is Array a && a.Length > 0 && a.GetValue(0) is Array) {
            var parts = new List<string>();
            foreach (var x in a) parts.Add(_fmt(x));
            return string.Join(" ", parts);
        }
        if (v is string s) return s;
        return _fmt(v);
    }
    public static string get_digits(long num_0) {
        long cube_1 = ((num_0 * num_0) * num_0);
        string s_2 = _fmtStr(cube_1);
        long[] counts_3 = new long[]{};
        long j_4 = 0;
        while ((j_4 < 10)) {
            counts_3 = (Enumerable.ToArray(Enumerable.Append<long>(counts_3, 0)));
            j_4 = (j_4 + 1);
        };
        long i_5 = 0;
        while ((i_5 < s_2.Length)) {
            long d_6 = _atoi(s_2.Substring((int)(i_5), 1));
            counts_3[d_6] = (counts_3[(int)(d_6)] + 1);
            i_5 = (i_5 + 1);
        };
        string result_7 = "";
        long d_8 = 0;
        while ((d_8 < 10)) {
            long c_9 = counts_3[(int)(d_8)];
            while ((c_9 > 0)) {
                result_7 = (result_7 + _fmtStr(d_8));
                c_9 = (c_9 - 1);
            }
            d_8 = (d_8 + 1);
        };
        return result_7;
    }

    public static long solution(long max_base_10) {
        Dictionary<string, long[]> freqs_11 = new Dictionary<string, long[]>{};
        long num_12 = 0;
        while (true) {
            string digits_13 = Program.get_digits(num_12);
            long[] arr_14 = new long[]{};
            if ((freqs_11.ContainsKey(digits_13))) {
                arr_14 = (freqs_11.ContainsKey(digits_13) ? freqs_11[digits_13] : null);
            }
            arr_14 = (Enumerable.ToArray(Enumerable.Append<long>(arr_14, num_12)));
            freqs_11[digits_13] = arr_14;
            if ((arr_14.Length == max_base_10)) {
                long base_15 = arr_14[(int)(0)];
                return ((base_15 * base_15) * base_15);
            }
            num_12 = (num_12 + 1);
        };
        return default(long);
    }

    static void Main() {
        {
            var __memStart = _mem();
            var __start = _now();
            Console.WriteLine(Program._fmtTop(("solution() = " + _fmtStr(Program.solution(5)))));
            var __end = _now();
            var __memEnd = _mem();
            var __dur = (__end - __start);
            if (__dur <= 0) __dur = 1;
            var __memDiff = __memEnd - __memStart;
            if (__memDiff <= 0) __memDiff = __memEnd;
            Console.WriteLine("{\"name\":\"main\",\"duration_us\":" + __dur + ",\"memory_bytes\":" + __memDiff + "}");
        }
    }
}
