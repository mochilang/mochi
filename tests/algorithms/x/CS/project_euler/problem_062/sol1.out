/tmp/TestCSTranspiler_Algorithms_Golden001_project_euler_problem_062_sol12752596524/001/Program.cs(158,9): warning CS0162: Unreachable code detected [/tmp/TestCSTranspiler_Algorithms_Golden001_project_euler_problem_062_sol12752596524/001/app.csproj]
solution() = 127035954683